### PR TITLE
#2947 - Display property names for non-object models

### LINKED ISSUE #2947: Title not displayed for non-object models
Models that are not objects (scalars, arrays) have title missing.

## Definition:
```
definitions:
  Scalar:
    type: string
  Array:
    type: array
    items:
      type: string
```
## Expected result:
```
Scalar ⇄ Scalar string
Array  ⇄ Array[string]
```
## Actual result:
```
string
[...]
```
[Ghist demostrating issue](https://gist.githubusercontent.com/lekoder/c03a0751609829888451909539f315be/raw/0be8864b836ae2aaaed18c01c3995bbfc0c5cbff/swagger-ui-scalar-title-missing.yaml)

Affected version: **v3.0.0-229-g7790383**

### PR DESCRIPTION
**Pull request by owenconti**
Fixes #2947.

I wasn't 100% sure how Models are supposed to look inside arrays and as properties as objects.

screen shot 2017-06-25 at 10 25 31 pm

### CODE REVIEW

**Comment by webron** [comment]
@owenconti can you share the definition you used for testing?

**Comment by owenconti** [comment]
```
swagger: '2.0'
info:
  title: Composing Error Demo
  description: demo of composing error
  version: 0.1.0
schemes:
  - http
paths:
  /:
    get:
      responses:
        '200':
          description: Dummy description
          schema:
            type: object
            properties:
              test: 
                $ref: '#/definitions/Object1'
definitions:
  Array1:
    type: "array"
    items:
      type: integer
      format: int32
      description: Position in pagination.
  String1:
    type: "string"
  Boolean1:
    type: "boolean"
  Object1:
    type: "object"
    properties:
      offset:
        type: integer
        format: int32
        description: Position in pagination.
      items:
        $ref: '#/definitions/Array1'
```

**Comment by webron** [comment]
I think it looks fine. There's a slight offset when a model title is displayed next to a property name, but that existed before this PR. Maybe @Minasokoni could take a look at that.

**Comment by Cheprer** [comment]
Thanks for the PR, looks alright but since I'm not a contributor it's not on me but I would appreciate this fix as well in the bundle once you merge this, cheers 👍

File: src/core/components/model.jsx [hunk 1]
```diff
@@ -176,17 +183,20 @@ class ArrayModel extends Component {
...
-      </span>
+      {
+        title && <span className="model-title">
+          <span className="model-title__text">{ title }</span>
+        </span>
+      }
       <Collapse collapsed={ depth > expandDepth } collapsedContent="[...]">
         [
-          <span><Model { ...this.props } schema={ items } required={ false }/></span>
+          <span><Model { ...this.props } name="" schema={ items } required={ false }/></span>
```
Comment (owenconti) [inline src/core/components/model.jsx]: It's definitely a bit of a hack. Here are screenshots showing the difference with `name=""` and without it.

Without `name=""`
screen shot 2017-06-28 at 6 09 00 pm

With `name=""`
screen shot 2017-06-28 at 6 09 23 pm

Comment (shockey) [inline src/core/components/model.jsx]: Ah, right, I see now. `{ ...this.props }` is sending in the name provided to ArrayModel, so you need to override that.

Seems ok to me, just wanted to make sure it had a purpose 😄

**Pull request merged**